core/src/main/java/com/graphhopper/GraphHopper.java

```diff
@@ -820,6 +820,10 @@ private void process(boolean closeEarly) {
     }
 
     protected void postImport() {
+        // Important note: To deal with via-way turn restrictions we introduce artificial edges in OSMReader (#2689).
+        // These are simply copies of real edges. Any further modifications of the graph edges must take care of keeping
+        // the artificial edges in sync with their real counterparts. So if an edge attribute shall be changed this change
+        // must also be applied to the corresponding artificial edge.
```

Comment:
a) could also allow graph sorting when turn costs are enabled, because the turn cost storage would be written only after the graph was sorted. But I'm not even sure the graph sorting is useful?

Reply:
Or would it be feasible to store some kind of 'reference' for one edge that points to the properties of another? This would save some memory (but not a lot), and more importantly the artificial edges would stay in sync with the real ones automatically.

Reply:
> Doing this via the KV storage would be fine as only a few selected edges would use this.

Yes, but it would only be useful if it could be used transparently, i.e. calling something like `edge.get(whateverEnc)` should automatically return the referenced edges value.

I'd say we keep this as it is for now.